tools/upgrade.sh

```diff
@@ -34,6 +34,12 @@ then
   printf "${BLUE}%s\n" "Hooray! Oh My Zsh has been updated and/or is at the current version."
   printf "${BLUE}${BOLD}%s${NORMAL}\n" "To keep up on the latest news and updates, follow us on twitter: https://twitter.com/ohmyzsh"
   printf "${BLUE}${BOLD}%s${NORMAL}\n" "Get your Oh My Zsh swag at:  http://shop.planetargon.com/"
+
+  if [ -f $ZSH_CUSTOM/tools/custom_event.sh ]; then
+    # Notify user's custom code of upgrades
+    env ZSH=$ZSH ZSH_CUSTOM=$ZSH_CUSTOM /bin/sh $ZSH_CUSTOM/tools/custom_event.sh upgrade
```

Comment:
It is where the customizer provides a script to consume events. For the **upgrade** event with a github based customization they would have some code to pull the latest changes. For example, my customizations live at: https://github.com/mscalora/custom-zsh so I would have a script that would look somthing like (just an example):

    #!/bin/sh
    pushd $ZSH_CUSTOM
    if git pull --rebase --stat origin master ; then
        echo "Updated custom folder"
    else
        echo "Error: couldn't update custom folder"
    fi
    popd